### DISCUSSION TITLE
ROS noetic Build Fails

### DISCUSSION THREAD

**Discussion by JeremiasGaia** [OpenVSLAM-Community/openvslam_ros#45] [Q&A] [answered]
Hi. My openVSLAM installation works fine.
However, recently i tried to install ROS packages following the instructions on https://openvslam-community.readthedocs.io/en/latest/ros_package.html#installation for ROS noetic.

When i run 
catkin_make -DUSE_PANGOLIN_VIEWER=ON -DUSE_SOCKET_PUBLISHER=OFF

I get the following error, I hope you can help me

Base path: /home/jeremias/OVS_catkin_ws
Source space: /home/jeremias/OVS_catkin_ws/src
Build space: /home/jeremias/OVS_catkin_ws/build
Devel space: /home/jeremias/OVS_catkin_ws/devel
Install space: /home/jeremias/OVS_catkin_ws/install
####
####
-- Using CATKIN_DEVEL_PREFIX: /home/jeremias/OVS_catkin_ws/devel
-- Using CMAKE_PREFIX_PATH: /opt/ros/noetic
-- This workspace overlays: /opt/ros/noetic
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.8.10", minimum required is "3") 
-- Using PYTHON_EXECUTABLE: /usr/bin/python3
-- Using Debian Python package layout
-- Using empy: /usr/lib/python3/dist-packages/em.py
-- Using CATKIN_ENABLE_TESTING: ON
-- Call enable_testing()
-- Using CATKIN_TEST_RESULTS_DIR: /home/jeremias/OVS_catkin_ws/build/test_results
-- Forcing gtest/gmock from source, though one was otherwise available.
-- Found gtest sources under '/usr/src/googletest': gtests will be built
-- Found gmock sources under '/usr/src/googletest': gmock will be built
-- Found PythonInterp: /usr/bin/python3 (found version "3.8.10") 
-- Using Python nosetests: /usr/bin/nosetests3
-- catkin 0.8.10
-- BUILD_SHARED_LIBS is on
-- BUILD_SHARED_LIBS is on
-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- ~~  traversing 2 packages in topological order:
-- ~~  - cv_bridge
-- ~~  - openvslam_ros
-- ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-- +++ processing catkin package: 'cv_bridge'
-- ==> add_subdirectory(cv_bridge)
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.8.so (found version "3.8.10") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython3.8.so (found suitable version "3.8.10", minimum required is "3.8") 
-- +++ processing catkin package: 'openvslam_ros'
-- ==> add_subdirectory(openvslam_ros)
-- Build type: Release
-- Address/Memory sanitizer: DISABLED
-- Stack trace logger: DISABLED
-- Google Perftools: DISABLED
-- Architecture-aware optimization (-march=native): DISABLED
-- C++11 support: OK (-std=c++17)
-- Using these message generators: gencpp;geneus;genlisp;gennodejs;genpy
CMake Error at openvslam_ros/CMakeLists.txt:134 (find_package):
  By not providing "Findopenvslam.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "openvslam", but CMake did not find one.

  Could not find a package configuration file provided by "openvslam" with
  any of the following names:

    openvslamConfig.cmake
    openvslam-config.cmake

  Add the installation prefix of "openvslam" to CMAKE_PREFIX_PATH or set
  "openvslam_DIR" to a directory containing one of the above files.  If
  "openvslam" provides a separate development package or SDK, be sure it has
  been installed.

-- Configuring incomplete, errors occurred!
See also "/home/jeremias/OVS_catkin_ws/build/CMakeFiles/CMakeOutput.log".
See also "/home/jeremias/OVS_catkin_ws/build/CMakeFiles/CMakeError.log".
Invoking "cmake" failed

**Comment by ymd-stella** [comment, marked as answer]
Read the error carefully. It means that you have not installed OpenVSLAM.

**Reply by JeremiasGaia** [reply]
Ok, by reading your comments I found that in the installation [guide](https://openvslam-community.readthedocs.io/en/latest/installation.html)  the command `make install` is missing. 
If you execute 
```
cd /path/to/openvslam
mkdir build && cd build
cmake \
    -DUSE_PANGOLIN_VIEWER=OFF \
    -DUSE_SOCKET_PUBLISHER=ON \
    -DINSTALL_SOCKET_PUBLISHER=ON \
    -DUSE_STACK_TRACE_LOGGER=ON \
    -DBUILD_TESTS=ON \
    -DBUILD_EXAMPLES=ON \
    ..
make -j4
```
the code is built, but not installed. Therefore ROS can't find it.
I suggest to add the make install command to the official guide:

```
cd /path/to/openvslam
mkdir build && cd build
cmake \
    -DUSE_PANGOLIN_VIEWER=OFF \
    -DUSE_SOCKET_PUBLISHER=ON \
    -DINSTALL_SOCKET_PUBLISHER=ON \
    -DUSE_STACK_TRACE_LOGGER=ON \
    -DBUILD_TESTS=ON \
    -DBUILD_EXAMPLES=ON \
    ..
make -j4
make install
```
thank you for your answer, I hope my feedback helps.
